# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/systemservices/android_video_capture/device_info_android.h
# Commit: 55e2a4c5cab9
# Full Hash: 55e2a4c5cab961f0fa8627a490c4d714584b3065
# Author: Dan Minor <dminor@mozilla.com>
# Date: 2020-02-11 21:47:36
# Regressor Bug: 1578073
# File Overlap Count: 2
# Description:
#   Bug 1578073 - Move android video capture code to dom/media/systemservices; r=jib
#   
#   Although originally part of webrtc.org, this code has subsequently been
#   removed by upstream. Moving it to under dom/media should make it clearer that
#   this is code that we are maintaining and simplify future upstream merges.
# ==============================================================================

diff -r 27b45ca46f24 -r 55e2a4c5cab9 dom/media/systemservices/android_video_capture/device_info_android.h
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/dom/media/systemservices/android_video_capture/device_info_android.h	Mon Feb 10 17:08:36 2020 +0000
@@ -0,0 +1,72 @@
+/*
+ *  Copyright (c) 2012 The WebRTC project authors. All Rights Reserved.
+ *
+ *  Use of this source code is governed by a BSD-style license
+ *  that can be found in the LICENSE file in the root of the source
+ *  tree. An additional intellectual property rights grant can be found
+ *  in the file PATENTS.  All contributing project authors may
+ *  be found in the AUTHORS file in the root of the source tree.
+ */
+
+#ifndef WEBRTC_MODULES_VIDEO_CAPTURE_MAIN_SOURCE_ANDROID_DEVICE_INFO_ANDROID_H_
+#define WEBRTC_MODULES_VIDEO_CAPTURE_MAIN_SOURCE_ANDROID_DEVICE_INFO_ANDROID_H_
+
+#include <jni.h>
+
+#include "modules/video_capture/device_info_impl.h"
+#include "modules/video_capture/video_capture_impl.h"
+
+#define AndroidJavaCaptureDeviceInfoClass \
+  "org/webrtc/videoengine/VideoCaptureDeviceInfoAndroid"
+#define AndroidJavaCaptureCapabilityClass \
+  "org/webrtc/videoengine/CaptureCapabilityAndroid"
+
+namespace webrtc {
+namespace videocapturemodule {
+
+class DeviceInfoAndroid : public DeviceInfoImpl {
+ public:
+  static void Initialize(JavaVM* javaVM);
+  static void DeInitialize();
+
+  DeviceInfoAndroid();
+  virtual ~DeviceInfoAndroid();
+
+  // Set |*index| to the index of the camera matching |deviceUniqueIdUTF8|, or
+  // return false if no match.
+  bool FindCameraIndex(const char* deviceUniqueIdUTF8, size_t* index);
+
+  virtual int32_t Init();
+  virtual uint32_t NumberOfDevices();
+  virtual int32_t Refresh();
+  virtual int32_t GetDeviceName(uint32_t deviceNumber, char* deviceNameUTF8,
+                                uint32_t deviceNameLength,
+                                char* deviceUniqueIdUTF8,
+                                uint32_t deviceUniqueIdUTF8Length,
+                                char* productUniqueIdUTF8 = 0,
+                                uint32_t productUniqueIdUTF8Length = 0,
+                                pid_t* pid = 0);
+  virtual int32_t CreateCapabilityMap(const char* deviceUniqueIdUTF8);
+
+  virtual int32_t DisplayCaptureSettingsDialogBox(
+      const char* /*deviceUniqueIdUTF8*/, const char* /*dialogTitleUTF8*/,
+      void* /*parentWindow*/, uint32_t /*positionX*/, uint32_t /*positionY*/) {
+    return -1;
+  }
+  virtual int32_t GetOrientation(const char* deviceUniqueIdUTF8,
+                                 VideoRotation& orientation);
+
+  // Populate |min_mfps| and |max_mfps| with the closest supported range of the
+  // device to |max_fps_to_match|.
+  void GetMFpsRange(const char* deviceUniqueIdUTF8, int max_fps_to_match,
+                    int* min_mfps, int* max_mfps);
+
+ private:
+  enum { kExpectedCaptureDelay = 190 };
+  static void BuildDeviceList();
+};
+
+}  // namespace videocapturemodule
+}  // namespace webrtc
+
+#endif  // WEBRTC_MODULES_VIDEO_CAPTURE_MAIN_SOURCE_ANDROID_DEVICE_INFO_ANDROID_H_